sky/backends/cloud_vm_ray_backend.py

```diff
@@ -657,6 +657,15 @@ def _update_blocklist_on_gcp_error(
                     # {'code': 8, 'message': 'There is no more capacity in the zone "europe-west4-a"; you can try in another zone where Cloud TPU Nodes are offered (see https://cloud.google.com/tpu/docs/regions) [EID: 0x1bc8f9d790be9142]'} # pylint: disable=line-too-long
                     self._blocked_resources.add(
                         launchable_resources.copy(zone=zone.name))
+                elif code == 'RESOURCE_NOT_FOUND':
+                    # https://github.com/skypilot-org/skypilot/issues/1797
+                    # The VM may be alive on console. In the inner provision
+                    # loop we have used retries to recover but failed. The
+                    # provision loop will terminate the potentially live VMs
+                    # and move onto the next zone. Since the VM may have been
+                    # provisioned in this zone, it doesn't seem right to block
+                    # the current zone.
+                    pass
```

Comment:
Since we have done the retry in `_gang_schedule_ray_up` by adding that error in the `need_ray_up`, we should add the zone to the block list here, to avoid the outer failover loop go to that zone again.

Reply:
IIRC, without `-r` the outer loop will do a single pass over all zones, so this zone will not be tried again. So blocking vs. not blocking has the same effect. With `-r`, it's desirable to retry this zone. Is this the case?

Reply:
Our yield zones will not loop through all the zones in a region, as we currently rely on the optimizer to generate optimization result for per-zone. That is to say the failover for zones will happen [here](https://github.com/skypilot-org/skypilot/blob/2f5fa21b1d0b1e537ae03a845e214b454a17ad5e/sky/backends/cloud_vm_ray_backend.py#L1795-L1799), but not [here](https://github.com/skypilot-org/skypilot/blob/2f5fa21b1d0b1e537ae03a845e214b454a17ad5e/sky/backends/cloud_vm_ray_backend.py#L1171).
However, since we have add the `to_provision` in the block list [here](https://github.com/skypilot-org/skypilot/blob/2f5fa21b1d0b1e537ae03a845e214b454a17ad5e/sky/backends/cloud_vm_ray_backend.py#L1769), it should explain why it does not go to that zone again. I still prefer to add that to the block list here for consistency with the other parts.

Reply:
Oh, that's right. I got caught up thinking `_yield_zones()` will go through the zones. Added now.